inc/dropdown.class.php

```diff
@@ -2631,193 +2676,255 @@ public static function getDropdownValue($post, $json = true) {
-                              ON (`glpi_softwarelicenses`.`softwares_id` = `glpi_softwares`.`id`)
-                        $where";
+               $criteria = [
+                  'SELECT' => [
+                     "$table.*",
+                     new \QueryExpression("CONCAT(glpi_softwares.name,' - ',glpi_softwarelicenses.name) AS $field")
+                  ],
+                  'FROM'   => $table,
+                  'LEFT JOIN' => [
+                     'glpi_softwares'  => [
+                        'IN' => [
+                           'glpi_softwarelicenses' => 'softwares_id',
+                           'glpi_softwares'        => 'id'
+                        ]
+                     ]
+                  ]
+               ];
                break;
 
             case "Profile" :
-               $query = "SELECT DISTINCT `$table`.*
-                        FROM `$table`
-                        LEFT JOIN `glpi_profilerights`
-                              ON (`glpi_profilerights`.`profiles_id` = `$table`.`id`)
-                        $where";
+               $criteria = [
+                  'SELECT DISTINCT' => "$table.*",
+                  'FROM'            => $table,
+                  'LEFT JOIN'       => [
+                     'glpi_profilerights' => [
+                        'ON' => [
+                           'glpi_profilerights' => 'profiles_id',
+                           $table               => 'id'
+                        ]
+                     ]
+                  ]
+               ];
                break;
 
             case KnowbaseItem::getType():
-               $query = "SELECT DISTINCT `$table`.* $addselect
-                        FROM `$table`
-                        $addjoin ".KnowbaseItem::addVisibilityJoins()."
-                        $where";
+               $criteria = [
+                  'SELECT DISTINCT' => "$table.*",
+                  'FIELDS'          => $addselect,
+                  'FROM'            => $table
+               ];
+               if (count($ljoin)) {
+                  $criteria['LEFT JOIN'] = $ljoin;
+               }
+
+               $visibility = KnowbaseItem::getVisibilityCriteria();
+               if (count($visibility['LEFT JOIN'])) {
+                  $criteria['LEFT JOIN'] = array_merge($criteria['LEFT JOIN'], $visibility['LEFT JOIN']);
+                  //Do not use where??
+                  /*if (isset($visibility['WHERE'])) {
+                     $where = $visibility['WHERE'];
+                  }*/
+               }
                break;
 
             case Project::getType():
-               $addjoin .= Project::addVisibilityJoins();
-               $where   .= Project::addVisibility();
+               $visibility = Project::getVisibilityCriteria();
+               if (count($visibility['LEFT JOIN'])) {
+                  $ljoin[] = $visibility['LEFT JOIN'];
+                  if (isset($visibility['WHERE'])) {
```

Comment:
Is it possible to have `WHERE` without `LEFT JOIN` ?

Reply:
Nope, the inverse, see comment line 2790